### DISCUSSION TITLE
re: Contributing to the repo

### DISCUSSION THREAD

**Discussion by GRibbans** [bigfug/Fugio#87] [Ideas]
Hi @bigfug, All :). After coming across Fugio the other day, and then your YT videos and comments, I was looking at ways I could pitch in and help. What is the best way for me to do this? Pull requests and make my edits and Commit for review? Or detail them here, or over email, or message on Discord?

**Comment by bigfug** [comment]
Hi @GRibbans, thanks for the offer of help - very much appreciated! It would depend on the kind of thing you'd want to get stuck into - there's a lot of parts!

**I'm working exclusively on the 4.0 development branches at the moment.**

Bug fixes are very welcome with pull/commits. Adding features, new nodes, pins, etc to the existing plugins should be discussed first just so their not doing something that's due to be deprecated or implemented elsewhere.

Creating your own separate plugins doesn't need my input, obviously! Although it's worth checking with me to see if I've done any work already - I have a bunch of unreleased plugins. The interface is pretty much defined, but I am making a few changes in 4.0.

**On the main repo:**

If you're a CMAKE person, the build system could probably do with another set of eyes on it to see if there are better ways to do things, especially with the new distributed plugin system.

If you are a GitHub Actions master, the CI/CD for multiple platforms needs looking at, specifically building for macOS and Linux. I'll set up a Raspberry Pi remote build system at some point.

Generally building on macOS and Linux - I'm personally just focussing on Windows at the moment. 

Documentation. Both for users and developers. The biggest help would be thinking about what documentation is needed, and how to structure it. I'm happy to write it, but I need to know what people need to know! The idea is to put it all on the GitHub wiki.

I'm also keen for input on the user interface, both on how to make it look much better, as well layout ideas.

I'd like to replace the OpenGL plugin with one that exclusively uses the Qt3D interface, which internally supports Vulkan, Metal, and OpenGL. I think the whole 3D geometry/shader pipeline probably needs a redesign to be much more user friendly.

Basically loads to do, I'd just ask you check in with me before pouring hours of your life into something major!
